lib/ical/helpers.js

```diff
@@ -1,377 +1,314 @@
-        pos += 1;
-      } else {
-        return pos;
-      }
+/**
+ * Identical to indexOf but will only match values when they are not preceded
+ * by a backslash character.
+ *
+ * @param {String} buffer         String to search
+ * @param {String} search         Value to look for
+ * @param {Number} pos            Start position
+ * @return {Number}               The position, or -1 if not found
+ */
+export function unescapedIndexOf(buffer, search, pos) {
+  while ((pos = buffer.indexOf(search, pos)) !== -1) {
+    if (pos > 0 && buffer[pos - 1] === '\\') {
+      pos += 1;
+    } else {
+      return pos;
     }
-    return -1;
-  },
+  }
+  return -1;
+}
 
-  /**
-   * Find the index for insertion using binary search.
-   *
-   * @param {Array} list            The list to search
-   * @param {?} seekVal             The value to insert
-   * @param {function(?,?)} cmpfunc The comparison func, that can
-   *                                  compare two seekVals
-   * @return {Number}               The insert position
-   */
-  binsearchInsert: function(list, seekVal, cmpfunc) {
-    if (!list.length)
-      return 0;
+/**
+ * Find the index for insertion using binary search.
+ *
+ * @param {Array} list            The list to search
+ * @param {?} seekVal             The value to insert
+ * @param {function(?,?)} cmpfunc The comparison func, that can
+ *                                  compare two seekVals
+ * @return {Number}               The insert position
+ */
+export function binsearchInsert(list, seekVal, cmpfunc) {
+  if (!list.length)
+    return 0;
 
-    var low = 0, high = list.length - 1,
-        mid, cmpval;
+  let low = 0, high = list.length - 1,
+      mid, cmpval;
 
-    while (low <= high) {
-      mid = low + Math.floor((high - low) / 2);
-      cmpval = cmpfunc(seekVal, list[mid]);
-
-      if (cmpval < 0)
-        high = mid - 1;
-      else if (cmpval > 0)
-        low = mid + 1;
-      else
-        break;
-    }
+  while (low <= high) {
+    mid = low + Math.floor((high - low) / 2);
+    cmpval = cmpfunc(seekVal, list[mid]);
 
     if (cmpval < 0)
-      return mid; // insertion is displacing, so use mid outright.
+      high = mid - 1;
     else if (cmpval > 0)
-      return mid + 1;
+      low = mid + 1;
     else
-      return mid;
-  },
-
-  /**
-   * Convenience function for debug output
-   * @private
-   */
-  dumpn: /* istanbul ignore next */ function() {
-    if (!ICAL.debug) {
-      return;
-    }
-
-    if (typeof (console) !== 'undefined' && 'log' in console) {
-      ICAL.helpers.dumpn = function consoleDumpn(input) {
-        console.log(input);
-      };
-    } else {
-      ICAL.helpers.dumpn = function geckoDumpn(input) {
-        dump(input + '\n');
-      };
-    }
+      break;
+  }
 
-    ICAL.helpers.dumpn(arguments[0]);
-  },
+  if (cmpval < 0)
+    return mid; // insertion is displacing, so use mid outright.
+  else if (cmpval > 0)
+    return mid + 1;
+  else
```

Comment:
All `else` after `return`s can be deleted.

Reply:
Technically yes, aesthetically I like having them sometimes. I'd rather handle this via linter, but am personally not keen on making the change (the lack of brackets here triggers me more, but that should also be a linter change :)

This PR is already big enough, lets defer this.